pkg/compiler/codegen.go

```diff
@@ -1828,27 +1828,17 @@ func (c *codegen) convertStruct(lit *ast.CompositeLit, ptr bool) {
 	}
 
 	emit.Opcodes(c.prog.BinWriter, opcode.NOP)
-	emit.Int(c.prog.BinWriter, int64(strct.NumFields()))
-	if ptr {
-		emit.Opcodes(c.prog.BinWriter, opcode.NEWARRAY)
-	} else {
-		emit.Opcodes(c.prog.BinWriter, opcode.NEWSTRUCT)
-	}
-
 	keyedLit := len(lit.Elts) > 0
 	if keyedLit {
 		_, ok := lit.Elts[0].(*ast.KeyValueExpr)
 		keyedLit = keyedLit && ok
 	}
 	// We need to locally store all the fields, even if they are not initialized.
 	// We will initialize all fields to their "zero" value.
-	for i := 0; i < strct.NumFields(); i++ {
+	for i := strct.NumFields() - 1; i >= 0; i-- {
```

Comment:
Initially I thought about the evaluation order, but I found no mentions of structs here https://golang.org/ref/spec#Order_of_evaluation. I think they should behave similarly to other types, so this is ok.